crates/message_broker/src/events_queue.rs

```diff
@@ -1,15 +1,15 @@
 use crate::redis::Redis;
 use async_trait::async_trait;
 use deadpool_redis::redis::AsyncCommands;
-use quests_definitions::{quests::*, ProstMessage};
+use quests_definitions::ProstMessage;
 use std::sync::Arc;
 
 pub type EventsQueueResult<T> = Result<T, String>;
 
 #[async_trait]
-pub trait EventsQueue: Send + Sync {
-    async fn push(&self, event: &Event) -> EventsQueueResult<usize>;
-    async fn pop(&self) -> EventsQueueResult<Event>;
+pub trait EventsQueue<T>: Send + Sync {
```

Comment:
Since it's now generic, I think we should rename the trait and implementation struct